tests/test_engines.py

```diff
@@ -33,17 +32,29 @@ def test_default_pull_source_valid_url(self):
             assert isinstance(response, Webpage)
             assert isinstance(response.source, str)
 
+    def test_curried_function(self):
+        """
+        Changes to the state of the Engine instance after currying shouldn't
+        affect the parameters of the curried method
+        """
+        self.engine.headers.update({'Content-Type': 'text/json'})
+        get_source = self.engine.curry()
+        self.engine.headers.update({'Content-Type': 'text/xml'})
+        page1 = get_source('http://127.0.0.1:5000/echo-headers/')
```

Comment:
OK check out the latest commit - removed flask and switched to using httpbin. I had to refactor the get_page_source() method a little because it wasn't adding the instance method's headers to the request unless headers were passed to the method. Still haven't taken care of the other issues but I'll get to them soon